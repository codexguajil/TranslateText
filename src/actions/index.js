export const storeTranslation = (translation) => ({
  type: 'STORE_TRANS',
  translation
})

export const deleteTrans = (id) => ({
  type: 'DELETE_TRANS',
  id
})

export const postMessage = (message) => ({
  type: 'MESSAGE',
  message
})