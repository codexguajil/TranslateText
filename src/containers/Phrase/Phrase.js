import React, { Component } from 'react'
import { connect } from 'react-redux'

export class Phrase extends Component {
  constructor(props) {
    super(props);
    this.state = {

    }
  }

  addFave = () => {
    
  }

  deletePhrase = () => {

  }

  render() {
    return (
      <div onClick={this.addFave}
           className='text-box'
      >{this.props.translatedText}
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  translations: state.translations
})

const mapDispatchToProps = (dispatch) => ({

})

export default connect(mapStateToProps)(Phrase)