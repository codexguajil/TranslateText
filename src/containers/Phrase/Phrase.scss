.text-box {
  background: linear-gradient(#FFF, rgb(179, 129, 179));
  border-radius: 10px;
  box-shadow: 0 5px 10px rgba(0,0,0,0.20);
  color: black;
  font-size: 26px;
  height: 180px;
  letter-spacing: 1px;
  margin: 30px;
  margin-bottom: 0;
  opacity: 0.9;
  padding: 8px;
  text-align: left;
  font-weight: 300;
  overflow: scroll;
  width: 450px;
  &:hover {
    background-color: linear-gradient(rgb(179, 129, 179), #FFF);
    cursor: pointer;
    opacity: 1;
    transition: background-color .5s ease;
  }
}

.text-link {
  text-decoration: none;
}

.delete {
  border: 1px dotted cadetblue;
  color: cadetblue;
  font-size: 16px;
  height: 20px;
  margin-top: 15px;
  line-height: 1.2;
  width: 100px;
  border-radius: 80px;
  &:hover {
    height: 50px;
    margin-top: 8px;
    width: 100px;
    box-shadow: 1px 1px black;
    cursor: pointer;
    transition: background-color 1s ease-out,
                width 1s ease,
                height .5s ease-in-out,
                margin-top 1s ease,
                border-radius 1s ease-in;
  }
  &:focus {
    outline: none;
  }
}

.phrase-box {
  height: 180px;
  margin: 44px;
}