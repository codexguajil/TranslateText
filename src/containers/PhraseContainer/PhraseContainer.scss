.container {
  background: rgba(0, 255, 255, 0.5);
  border-top: 1px solid black;
  display: flex;
  height: 800px;
  flex-wrap: wrap;
  background-image: linear-gradient(rgba(19, 223, 223, 0.5), #FFF);
  margin: auto;
  width: 100%;
}

#heading {
  border-bottom: 1px dotted #FFF;
  color: black;
  font-weight: 300;
  width: 400px;
}

.header-a {
  color: black;
  font-size: 32px;
  margin-top: 10px;
  padding: 10px;
  text-decoration: none;
  &:hover {
    color: grey;
  }
}

.banner {
  display: inline-flex;
  justify-content: space-between;
  width: 100%;
}

.loading {
  color: rgb(179, 129, 179);
  font-size: 60px;
}

#span {
  font-size: 16px;
}