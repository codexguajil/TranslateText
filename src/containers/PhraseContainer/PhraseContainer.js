import React, { Component } from 'react'
import { connect } from 'react-redux'
import Phrase from '../Phrase/Phrase'
import PropTypes from 'prop-types'
import {deleteTrans} from '../../actions'
import { Link, withRouter } from 'react-router-dom'

export class PhraseContainer extends Component {

  deleteTrans = (id, e) => {
    this.props.deleteTrans(id)
  }

  render() {
    return (
      <div>
        <div className='banner'>
          <h1 id="heading">Your Current Translations</h1>
          <Link to="/" className="header-a">
              Translate to French.
          </Link>
        </div>
      <div className="container">
        {this.props.loading && 
          <h1 className="loading">Loading...</h1>
        }
        {this.props.phrases && 
          this.props.phrases.map(phrase => <Phrase key={phrase.id} {...phrase} deleteTrans={this.deleteTrans}/>)
        }
      </div>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  phrases: state.translations
})

export const mapDispatchToProps = (dispatch) => ({
  deleteTrans: (id) => dispatch(deleteTrans(id))
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(PhraseContainer))

PhraseContainer.propTypes = {
  phrases: PropTypes.array
}