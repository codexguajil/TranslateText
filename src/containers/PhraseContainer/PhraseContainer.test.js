import React from 'react'
import {PhraseContainer, mapStateToProps, mapDispatchToProps} from './PhraseContainer'
import {Phrase} from '../Phrase/Phrase'
import {shallow} from 'enzyme'
import {deleteTrans} from '../../actions'
import {Link} from 'react-router-dom'

describe('PhraseContainer', () => {
  let wrapper;
    it('should match the snapshot', () => {
      wrapper = shallow(
        <PhraseContainer />
      )
      expect(wrapper).toMatchSnapshot()
    })

    it('should return a Phrase component for each phrase in store', () => {
      let props = {
        phrases: [{translatedText:'au revoir', id: '1'}, {translatedText: 'bonjour', id: '2'}],
        isLoading: false,
        children: []
      }

      wrapper = shallow(
        <PhraseContainer {...props} isLoading="false" deleteTrans={jest.fn}/>
      )

      let mockReturn = <div>
      <div className='banner'>
        <h1 id="heading">Your Current Translations</h1>
        <Link to="/" className="header-a">
            Translate to French.
        </Link>
      </div>
    <div className="container">
      {(wrapper.instance().props.isLoading) && 
        <h1 className="loading">Loading...</h1>
      }
      {(wrapper.instance().props.phrases) && 
        (wrapper.instance().props.phrases).map(phrase => <Phrase key={phrase.id} {...phrase} deleteTrans={wrapper.instance().deleteTrans}/>)
      }
    </div>
    </div>

      let expected = wrapper.instance().render()
      // expect(JSON.stringify(expected)).toEqual(JSON.stringify(mockReturn) )
    })

  describe('mapStateToProps', () => {
    it('should return the phrases in store', () => {
      const mockState = {
        translations: [],
        items: {}
      }

      const expected = {
        phrases: []
      }

      const mockProps = mapStateToProps(mockState)
      expect(mockProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with a deleteTrans action', () => {
      const mockId = 1

      const mockDispatch = jest.fn()
      const actionToDispatch = deleteTrans(mockId)

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.deleteTrans(mockId)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })

  describe('deleteTrans', () => {
    it('should delete a translation from the store', () => {
      let props = {
        deleteTrans: jest.fn()
      }
      let mockId = 1

      wrapper = shallow(
        <PhraseContainer {...props} />
      )

      wrapper.instance().deleteTrans(mockId)
      expect(wrapper.instance().props.deleteTrans).toHaveBeenCalledWith(mockId)
    })
  })
})