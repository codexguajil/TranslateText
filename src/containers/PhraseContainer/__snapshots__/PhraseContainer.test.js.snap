// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhraseContainer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PhraseContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="banner"
        >
          <div>
            <h1
              id="heading"
            >
              Your Current Translations
            </h1>
            <p>
              (click each one for details)
            </p>
          </div>
          <Link
            className="header-a"
            to="/"
          >
            Translate to French.
          </Link>
        </div>,
        <div
          className="container"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              <h1
                id="heading"
              >
                Your Current Translations
              </h1>
              <p>
                (click each one for details)
              </p>
            </div>,
            <Link
              className="header-a"
              to="/"
            >
              Translate to French.
            </Link>,
          ],
          "className": "banner",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  id="heading"
                >
                  Your Current Translations
                </h1>,
                <p>
                  (click each one for details)
                </p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Your Current Translations",
                  "id": "heading",
                },
                "ref": null,
                "rendered": "Your Current Translations",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "(click each one for details)",
                },
                "ref": null,
                "rendered": "(click each one for details)",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Translate to French.",
              "className": "header-a",
              "to": "/",
            },
            "ref": null,
            "rendered": "Translate to French.",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            undefined,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          undefined,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="banner"
          >
            <div>
              <h1
                id="heading"
              >
                Your Current Translations
              </h1>
              <p>
                (click each one for details)
              </p>
            </div>
            <Link
              className="header-a"
              to="/"
            >
              Translate to French.
            </Link>
          </div>,
          <div
            className="container"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                <h1
                  id="heading"
                >
                  Your Current Translations
                </h1>
                <p>
                  (click each one for details)
                </p>
              </div>,
              <Link
                className="header-a"
                to="/"
              >
                Translate to French.
              </Link>,
            ],
            "className": "banner",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1
                    id="heading"
                  >
                    Your Current Translations
                  </h1>,
                  <p>
                    (click each one for details)
                  </p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Your Current Translations",
                    "id": "heading",
                  },
                  "ref": null,
                  "rendered": "Your Current Translations",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "(click each one for details)",
                  },
                  "ref": null,
                  "rendered": "(click each one for details)",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Translate to French.",
                "className": "header-a",
                "to": "/",
              },
              "ref": null,
              "rendered": "Translate to French.",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              undefined,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
